routing/endpointregistry_test.go

```diff
@@ -0,0 +1,74 @@
+package routing
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEmptyRegistry(t *testing.T) {
+	r := NewEndpointRegistry(RegistryOptions{})
+	m := r.GetMetrics("some key")
+
+	assert.Equal(t, time.Time{}, m.DetectedTime())
+	assert.Equal(t, int64(0), m.InflightRequests())
+}
+
+func TestSetAndGet(t *testing.T) {
+	r := NewEndpointRegistry(RegistryOptions{})
+
+	mBefore := r.GetMetrics("some key")
+	r.IncInflightRequest("some key")
+	mAfter := r.GetMetrics("some key")
+
+	assert.Equal(t, int64(0), mBefore.InflightRequests())
+	assert.Equal(t, int64(1), mAfter.InflightRequests())
+
+	ts, _ := time.Parse(time.DateOnly, "2023-08-29")
+	mBefore = r.GetMetrics("some key")
+	r.SetDetectedTime("some key", ts)
+	mAfter = r.GetMetrics("some key")
+
+	assert.Equal(t, time.Time{}, mBefore.DetectedTime())
+	assert.Equal(t, ts, mAfter.DetectedTime())
+}
+
+func TestSetAndGetAnotherKey(t *testing.T) {
+	r := NewEndpointRegistry(RegistryOptions{})
+
+	r.IncInflightRequest("some key")
+	mToChange := r.GetMetrics("some key")
+	mConst := r.GetMetrics("another key")
+
+	assert.Equal(t, int64(0), mConst.InflightRequests())
+	assert.Equal(t, int64(1), mToChange.InflightRequests())
+}
+
+func benchmarkIncInflightRequests(b *testing.B, name string, goroutines int) {
+	const key string = "some key"
+
+	b.Run(name, func(b *testing.B) {
+		r := NewEndpointRegistry(RegistryOptions{})
+		wg := sync.WaitGroup{}
+		for i := 0; i < goroutines; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				for n := 0; n < b.N/goroutines; n++ {
+					r.IncInflightRequest(key)
+				}
+			}()
```

Comment:
I am not sure if the number of goroutines will matter here a lot. It could be that it runs one after the other without a lot of contention.